homeassistant/components/sensor/gitlab_ci.py

```diff
@@ -0,0 +1,182 @@
+"""Module for retrieving latest GitLab CI job information."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_SCAN_INTERVAL, CONF_TOKEN, CONF_URL)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+CONF_GITLAB_ID = 'gitlab_id'
+CONF_ATTRIBUTION = "Information provided by https://gitlab.com/"
+
+ICON_HAPPY = 'mdi:emoticon-happy'
+ICON_SAD = 'mdi:emoticon-happy'
+ICON_OTHER = 'mdi:git'
+
+ATTR_BUILD_ID = 'build id'
+ATTR_BUILD_STATUS = 'build_status'
+ATTR_BUILD_STARTED = 'build_started'
+ATTR_BUILD_FINISHED = 'build_finished'
+ATTR_BUILD_DURATION = 'build_duration'
+ATTR_BUILD_COMMIT_ID = 'commit id'
+ATTR_BUILD_COMMIT_DATE = 'commit date'
+ATTR_BUILD_BRANCH = 'master'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_GITLAB_ID): cv.string,
+    vol.Optional(CONF_URL, default='https://gitlab.com'): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL,
+                 default=timedelta(seconds=300)): cv.time_period,
+})
+
+REQUIREMENTS = ['python-gitlab==1.6.0']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Sensor platform setup."""
+    _priv_token = config.get(CONF_TOKEN)
```

Comment:
Use `dict[key]` for required config keys.

Reply:
Changed.